admin/asset_files/forms.py

```diff
@@ -7,6 +7,15 @@
 class ProviderAssetFileForm(forms.ModelForm):
     class Meta:
         model = ProviderAssetFile
-        fields = ['name', 'file', 'providers']
+        fields = ['name', 'file', 'providers', 'id']
 
+    id = forms.IntegerField(required=False, widget=forms.HiddenInput())
     providers = forms.ModelMultipleChoiceField(AbstractProvider.objects.all(), widget=forms.CheckboxSelectMultiple(), required=False)
+
+    def clean(self):
+        cleaned_data = super(ProviderAssetFileForm, self).clean()
+        obj_id = int(cleaned_data.get('id', None) or 0)
```

Comment:
Are there null values possible from the get other than the default?
Any reason this isn't just:
`int(cleaned_data.get('id', 0))`

Reply:
On create, `id` comes in as`None` because it's auto-generated later by the postgres sequence for the table. `{'id': None}.get('id', 0)` results in `None`, hence the conditional.